graphdatascience/procedure_surface/api/kcore_endpoints.py

```diff
@@ -0,0 +1,286 @@
+            The property name that contains weight
+
+        Returns
+        -------
+        DataFrame
+            DataFrame with the algorithm results containing nodeId and coreValue
+        """
+        pass
+
+    @abstractmethod
+    def write(
+        self,
+        G: Graph,
+        write_property: str,
+        relationship_types: Optional[List[str]] = None,
+        node_labels: Optional[List[str]] = None,
+        sudo: Optional[bool] = None,
+        log_progress: Optional[bool] = None,
+        username: Optional[str] = None,
+        concurrency: Optional[Any] = None,
+        job_id: Optional[Any] = None,
+        target_nodes: Optional[Any] = None,
+        relationship_weight_property: Optional[str] = None,
+        write_concurrency: Optional[Any] = None,
+    ) -> KCoreWriteResult:
+        """
+        Executes the K-Core algorithm and writes the results to the Neo4j database.
+
+        Parameters
+        ----------
+        G : Graph
+            The graph to run the algorithm on
+        write_property : str
+            The property name to write core values to
+        relationship_types : Optional[List[str]], default=None
+            The relationships types considered in this algorithm run
+        node_labels : Optional[List[str]], default=None
+            The node labels used to select nodes for this algorithm run
+        sudo : Optional[bool], default=None
+            Override memory estimation limits
+        log_progress : Optional[bool], default=None
+            Whether to log progress
+        username : Optional[str], default=None
+            The username to attribute the procedure run to
+        concurrency : Optional[Any], default=None
+            The number of concurrent threads
+        job_id : Optional[Any], default=None
+            An identifier for the job
+        target_nodes : Optional[Any], default=None
+            Subset of nodes to compute the algorithm for
+        relationship_weight_property : Optional[str], default=None
+            The property name that contains weight
+        write_concurrency : Optional[Any], default=None
+            The number of concurrent threads during the write phase
+
+        Returns
+        -------
+        KCoreWriteResult
+            Algorithm metrics and statistics
+        """
+        pass
+
+    @abstractmethod
+    def estimate(
+        self,
+        G: Optional[Graph] = None,
+        projection_config: Optional[dict[str, Any]] = None,
+    ) -> EstimationResult:
+        """
+        Estimate the results based on the provided graph and configuration.
+
+        This abstract method is intended to be implemented in a subclass to provide
+        specific functionality for estimating outcomes based on a graph name and a
+        projection configuration object. The implementation should use the given
+        parameters to compute and return an appropriate estimation result.
```

Comment:
this would be user-facing in our API docs? (will have to check once we hook them up in the AuraGraphDataScience object)